src/app/clusters/ota-server/ota-server.cpp

```diff
@@ -82,5 +160,47 @@ bool emberAfOtaSoftwareUpdateServerClusterQueryImageCallback(
     /* TYPE WARNING: array array defaults to */ uint8_t * protocolsSupported, uint8_t * location, uint8_t clientCanConsent,
     chip::ByteSpan metadataForServer)
 {
-    return false;
+    EmberAfStatus status         = EMBER_ZCL_STATUS_SUCCESS;
+    chip::EndpointId endpointId  = emberAfCurrentEndpoint();
+    OTAServerDelegate * delegate = GetDelegate(endpointId);
+
+    if (SendStatusIfDelegateNull(endpointId))
+    {
+        return true;
+    };
+
+    ChipLogDetail(Zcl, "OTA Server received QueryImage");
+
+    // TODO: (#7112) change location size checking once CHAR_STRING is supported
+    const uint8_t locationLen = emberAfStringLength(location);
+    if (locationLen != kLocationParamLength)
+    {
+        ChipLogError(Zcl, "expected location length %" PRIu8 ", got %" PRIu8, locationLen, kLocationParamLength);
+        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_ARGUMENT);
+    }
+    else if (metadataForServer.size() > kMaxMetadataLen)
+    {
+        ChipLogError(Zcl, "metadata size %zu exceeds max %zu", metadataForServer.size(), kMaxMetadataLen);
+        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_ARGUMENT);
+    }
+
+    chip::ByteSpan locationSpan(location, locationLen);
+
+    status = delegate->HandleQueryImage(vendorId, productId, imageType, hardwareVersion, currentVersion, protocolsSupported,
+                                        locationSpan, clientCanConsent, metadataForServer);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfSendImmediateDefaultResponse(status);
+    }
+
+    return true;
+}
+
+void chip::app::clusters::OTAServer::SetDelegate(chip::EndpointId endpointId, OTAServerDelegate * delegate)
```

Comment:
Why are doing "using namespace chip::app::clusters" only to have to spell out the entire thing here?

Can't we put the entire file content within that namespace?

Reply:
The entire file should not be in that namespace because the `emberAf...` functions are defined in the global namespace.

But I did remove the unnecessary scopes here